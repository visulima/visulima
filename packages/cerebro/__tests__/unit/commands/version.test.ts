import { describe, expect, it, vi } from "vitest";

import type { Cli as ICli } from "../../../src/@types/cli";
import type { Toolbox as IToolbox } from "../../../src/@types/toolbox";
import versionCommand from "../../../src/commands/version-command";

describe("command/version", () => {
    it("should output the version number when version is defined", async () => {
        expect.assertions(2);

        const loggerMock = {
            debug: vi.fn(),
            info: vi.fn(),
            warning: vi.fn(),
        };
        const runtimeMock = {
            getPackageVersion: vi.fn().mockReturnValue("1.0.0"),
        };

        await versionCommand.execute({ logger: loggerMock as unknown as Console, runtime: runtimeMock as unknown as ICli } as IToolbox);

        expect(runtimeMock.getPackageVersion).toHaveBeenCalledExactlyOnceWith();
        expect(loggerMock).toMatchSnapshot();
    });

    it("should warn and debug when version is undefined", async () => {
        expect.assertions(2);

        const loggerMock = {
            debug: vi.fn(),
            info: vi.fn(),
            warn: vi.fn(),
        };
        const runtimeMock = {
            getPackageVersion: vi.fn().mockReturnValue(undefined),
        };

        await versionCommand.execute({ logger: loggerMock as unknown as Console, runtime: runtimeMock as unknown as ICli } as IToolbox);

        expect(runtimeMock.getPackageVersion).toHaveBeenCalledExactlyOnceWith();
        expect(loggerMock).toMatchSnapshot();
    });
});
