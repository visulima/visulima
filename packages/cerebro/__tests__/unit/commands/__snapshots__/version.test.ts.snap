// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`command/version > should output the version number when version is defined 1`] = `
{
  "debug": [MockFunction],
  "info": [MockFunction] {
    "calls": [
      [
        "1.0.0",
      ],
    ],
    "results": [
      {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "warning": [MockFunction],
}
`;

exports[`command/version > should warn and debug when version is undefined 1`] = `
{
  "debug": [MockFunction] {
    "calls": [
      [
        "The version number was not provided by the cli constructor.",
      ],
    ],
    "results": [
      {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "info": [MockFunction],
  "warn": [MockFunction] {
    "calls": [
      [
        "Unknown version",
      ],
    ],
    "results": [
      {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;
